cdc/processor/sinkmanager/table_sink_wrapper.go

```diff
@@ -417,6 +419,28 @@ func (t *tableSinkWrapper) cleanRangeEventCounts(upperBound engine.Position, min
 	return shouldClean
 }
 
+func (t *tableSinkWrapper) sinkMaybeStuck(stuckCheck time.Duration) (bool, uint64) {
+	t.getCheckpointTs()
+
+	t.tableSink.RLock()
+	t.tableSink.innerMu.Lock()
+	version := t.tableSink.version
+	resolvedTs := t.tableSink.resolvedTs
+	checkpointTs := t.tableSink.checkpointTs
+	advanced := t.tableSink.advanced
+	t.tableSink.RUnlock()
+	t.tableSink.innerMu.Unlock()
+
+	// What these conditions mean:
+	// 1. the table sink has been associated with a valid sink;
+	// 2. its checkpoint hasn't been advanced for a while;
+	// 3. its checkpoint is less than the emitted point.
+	if version > 0 && time.Since(advanced) > stuckCheck && checkpointTs.Less(resolvedTs) {
```

Comment:
How to deal with the situation where resolvedTs suddenly push forward after being stuck for a while?

Reply:
Perhaps when `resolvedTs == checkpointTs`, we should also update advanced at the same time as resolvedTs moves forward?